RNTester/js/PanResponderExample.js

```diff
@@ -10,29 +10,32 @@
 
 'use strict';
 
-var React = require('react');
-var createReactClass = require('create-react-class');
-var ReactNative = require('react-native');
-var {PanResponder, StyleSheet, View} = ReactNative;
+const React = require('react');
+const ReactNative = require('react-native');
+const {PanResponder, StyleSheet, View} = ReactNative;
 
-var CIRCLE_SIZE = 80;
+import type {PanResponderInstance} from 'PanResponder';
 
-var PanResponderExample = createReactClass({
-  displayName: 'PanResponderExample',
+type CircleStylesType = {|
+  backgroundColor: string,
+  left: number,
+  top: number,
+|};
 
-  statics: {
-    title: 'PanResponder Sample',
-    description:
-      'Shows the use of PanResponder to provide basic gesture handling.',
-  },
+const CIRCLE_SIZE = 80;
+
+class PanResponderExample extends React.Component<{}> {
+  static title = 'PanResponder Sample';
+  static description =
+    'Shows the Use of PanResponder to provide basic gesture handling';
 
-  _panResponder: {},
-  _previousLeft: 0,
-  _previousTop: 0,
-  _circleStyles: {},
-  circle: (null: ?{setNativeProps(props: Object): void}),
+  _panResponder: PanResponderInstance | Object = {};
+  _previousLeft: number = 0;
+  _previousTop: number = 0;
+  _circleStyles: {|style: CircleStylesType | Object|} = {style: {}};
```

Comment:
The following should work if we make every property in `CircleStyleTypes` optional:

```JavaScript
 _circleStyles: {|style: CircleStylesType |} = {style: {}};
```

Reply:
👍